src/Persistence/NullItemPageLookup.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Persistence;
+
+use Title;
+use Wikibase\DataModel\Entity\ItemId;
+
+class NullItemPageLookup implements ItemPageLookup {
+
+	public function getPageTitle( ItemId $itemId ): ?Title {
+		return null;
+	}
```

Comment:
I wonder if the call to `rewriteLinkForNonEntityResult` in the hooks file is being skipped over in cases it should not when we always return null.

Reply:
`rewriteLinkForNonEntityResult` basically reimplements this: https://github.com/wikimedia/mediawiki-extensions-Wikibase/blob/3f636e441cf070234bfcd4f217603d2e2a0be28a/repo/includes/Hooks/HtmlPageLinkRendererEndHookHandler.php#L321-L327

When you override `$title` without `$titleSnippet` it ends up using `$title`'s page title for the link. So to get the original link text (derived from WB item) back, we need to trigger the original link text building. I couldn't get it to work by manually triggering the entire hook that calls that code.

So if `$pageTitle` is null, then we shouldn't do anything further, because the original entity link would already be correct as per the link building of Wikibase/WikibaseCirrusSearch.